packages/renderer/src/modules/store/deployment/slice.ts

```diff
@@ -126,6 +129,45 @@ export const executeDeployment = createAsyncThunk<
   }
 });
 
+export const executeDeploymentWithRetry = createAsyncThunk<
+  { info: Info; componentsStatus: DeploymentComponentsStatus },
+  string,
+  { rejectValue: { message: string; info: Info } }
+>('deployment/executeWithRetry', async (path, { dispatch, rejectWithValue, getState }) => {
+  const deployment = getState().deployment.deployments[path];
+  if (!deployment) {
+    return rejectWithValue({
+      message: 'Deployment not found. Initialize it first.',
+      info: {} as Info,
+    });
+  }
+
+  const { wallet, chainId } = deployment;
+  const triedServers = new Set<string>();
+
+  const result = await dispatchWithRetry(dispatch, executeDeployment, path, {
+    maxRetries: 3,
+    delayMs: 1000,
+    shouldRetry: (_error, attempt) => triedServers.size <= attempt,
+    onFailure: async () => {
+      const selectedServer = getAvailableCatalystServer(triedServers, chainId);
+      triedServers.add(selectedServer);
+
+      const port = await dispatch(publishScene({ path, target: selectedServer })).unwrap();
+
+      await dispatch(
+        initializeDeployment({
+          path,
+          port,
+          chainId,
+          wallet,
+        }),
+      ).unwrap();
+    },
+  });
+  return result;
+});
```

Comment:
careful with this one, because we only want to retry in case the catalyst rejects the deployment, and the implemetation of `executeDeployment` could also fail because the overall status is not a success. So if the LOD's fail to complete, the retry will be triggered with another catalyst when it shouldn't.

On another note, the `publishScene` function on `useEditor` hook does exactly what the `onFailure` callback is doing here. Maybe we can refactor that a little bit